lib/validator.ts

```diff
@@ -0,0 +1,101 @@
+import * as Axios from "axios";
+import { verify } from "jsonwebtoken";
+
+// tslint:disable-next-line:no-var-requires
+const jwkToPem = require("jwk-to-pem");
+
+import {
+    IJwtValidatorConfig,
+    IMapOfKidToPublicKey,
+    IPublicKeys,
+    ITokenHeader,
+    IClaim,
+    IValidationOptions,
+    IValidateTokenResult,
+} from "../declarations/declarations";
+import { DefaultValidationOptions } from "./constants";
+
+export class JwtValidator {
+    private validationOptions: IValidationOptions;
+    private cacheKeys?: IMapOfKidToPublicKey;
+
+    constructor(
+        private config: IJwtValidatorConfig,
+        validationOptions?: IValidationOptions,
+    ) {
+        this.validationOptions = Object.assign({}, DefaultValidationOptions, validationOptions || {});
+    }
+
+    public async validateToken<T>(token: string): Promise<IValidateTokenResult<T>> {
+        const result: IValidateTokenResult<T> = { isValid: false };
+        try {
+            const tokenSections = (token || "").split(".");
+            if (tokenSections.length < 2) {
```

Comment:
According to docs, shouldn't this be:
```suggestion
            if (tokenSections.length !== 3) {
```

Reply:
Yes, but in their example code is `< 2` 😄